src/main/java/com/minecolonies/coremod/client/render/TileEntityInfoPosterRenderer.java

```diff
@@ -0,0 +1,259 @@
+import com.minecolonies.blockout.Log;
+import com.minecolonies.coremod.tileentities.TileEntityInfoPoster;
+import net.minecraft.block.BlockWallSign;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.GuiUtilRenderComponents;
+import net.minecraft.client.model.ModelSign;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.VertexBuffer;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.world.World;
+import net.minecraftforge.client.model.IModel;
+import net.minecraftforge.client.model.ModelLoader;
+import net.minecraftforge.client.model.ModelLoaderRegistry;
+import net.minecraftforge.client.model.pipeline.LightUtil;
+import net.minecraftforge.common.model.IModelState;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import org.lwjgl.opengl.GL11;
+
+import java.util.List;
+
+@SideOnly(Side.CLIENT)
+public class TileEntityInfoPosterRenderer extends TileEntitySpecialRenderer<TileEntityInfoPoster>
+{
+    /**
+     * Offset to the block middle.
+     */
+    private static final double BLOCK_MIDDLE = 0.5;
+
+    /**
+     * Used to move the text slightly in the front.
+     */
+    private static final double SLIGHTLY_IN_FRONT = 0.01;
+
+    /**
+     * Scaling factor.
+     */
+    private static final double SCALING_FACTOR = 0.010416667F;
+
+    /**
+     * Y-Offset in order to have the scarecrow over ground.
+     */
+    private static final double YOFFSET    = 0.2;
+
+    /**
+     * 90° offset.
+     */
+    private static final int NINETY_DEGREE = 90;
+
+    /**
+     * 180° offset.
+     */
+    private static final int HALF_ROTATION = 180;
+
+    /**
+     * 270° offset.
+     */
+    private static final int THREE_QUARTERS  = 270;
+
+    /**
+     * Max text length.
+     */
+    private static final int MAX_TEXT_LENGTH = 90;
+
+    /**
+     * Text offset at x.
+     */
+    private static final int TEXT_OFFSET_X   = 10;
+
+    /**
+     * Text offset at y.
+     */
+    private static final int TEXT_OFFSET_Y   = 5;
+
+    /**
+     * The ModelSign instance for use in this renderer
+     */
+    private IBakedModel model = null;
+
+    @Override
+    public void renderTileEntityAt(final TileEntityInfoPoster te, final double x, final double y, final double z, final float partialTicks, final int destroyStage)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "renderTileEntityAt" is 16 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)